oshdb/src/main/java/org/heigit/ohsome/oshdb/impl/osh/OSHNodeImpl.java

```diff
@@ -73,16 +78,45 @@ public static OSHNodeImpl instance(final byte[] data, final int offset, final in
     // TODO maybe better to store number of versions instead
     final int dataLength = length - (dataOffset - offset);
 
-    return new OSHNodeImpl(data, offset, length, baseNodeId, baseTimestamp, baseLongitude, baseLatitude,
-        header, id, bbox, keys, dataOffset, dataLength);
+    return new OSHNodeImpl(data, offset, length, 
+        baseNodeId, baseTimestamp, baseLongitude, baseLatitude,
+        header, id, minLon, minLat, maxLon, maxLat, keys, dataOffset, dataLength);
   }
 
   private OSHNodeImpl(final byte[] data, final int offset, final int length, final long baseNodeId,
       final long baseTimestamp, final long baseLongitude, final long baseLatitude,
-      final byte header, final long id, final OSHDBBoundingBox bbox, final int[] keys,
+      final byte header, final long id, long minLon, long minLat, long maxLon, long maxLat, 
+      final int[] keys,
       final int dataOffset, final int dataLength) {
     super(data, offset, length, baseNodeId, baseTimestamp, baseLongitude, baseLatitude, header, id,
-        bbox, keys, dataOffset, dataLength);
+        minLon, minLat, maxLon, maxLat, keys, dataOffset, dataLength);
+    
+    // correct bbox!
```

Comment:
could you please state why this correcting is necessary? shouldn't this be even a case for throwing an exception (i.e. when the method is called with wrong parameters)?

Reply:
this is a remain of the current logic, as nodes don't necessary have a bbox(but a point in case of a single version) the bbox information is retrieved, by checking the versions it self.
The idea behind this is/was, to save space by not duplicating information.
But this code will be reworked soon.